packages/collector/test/tracing/logging/misc/test.js

```diff
@@ -19,74 +19,200 @@ mochaSuiteFn('tracing/logging/misc', function () {
+  describe('Ensure that logger spans are created', function () {
     beforeEach(async () => {
       await agentControls.clearReceivedTraceData();
       await agentControls.clearAgentLogs();
     });
 
-    describe('with the default logger', () => {
-      appControls.registerTestHooks({
-        instanaLoggingMode: 'uses-default-logger'
+    describe('with a custom pino logger', function () {
+      describe('set minimal pino configuration', function () {
+        let appControls;
+
+        beforeEach(async () => {
+          appControls = new AppControls({
+            instanaLoggingMode: 'receives-pino-logger',
+            env: {
+              CREATE_CUSTOM_LOG_SPANS: 'true'
+            }
+          });
+
+          await appControls.start();
+        });
+
+        afterEach(() => {
+          return appControls.stop();
+        });
+
+        it('log calls are traced', () => {
+          verifyLogSpansAreCreated(appControls, {
+            spanName: 'log.pino',
+            expectedTimeFormat: 'time',
+            expectedLevelFormat: 'level'
+          });
+        });
+      });
+
+      describe('set extended pino configuration', function () {
+        let appControls;
+
+        beforeEach(async () => {
+          appControls = new AppControls({
+            instanaLoggingMode: 'receives-pino-logger',
+            pipeSubprocessLogs: true,
+            env: {
+              CREATE_CUSTOM_LOG_SPANS: 'true',
+              EXTENDED_LOGGER_CONFIG: 'true'
+            }
+          });
+
+          await appControls.start();
+        });
+
+        afterEach(() => {
+          return appControls.stop();
+        });
+
+        it('log calls are traced', () =>
+          verifyLogSpansAreCreated(appControls, {
+            spanName: 'log.pino',
+            expectedTimeFormat: '"@timestamp":',
+            expectedLevelFormat: '"log.level":'
+          }));
       });
+    });
+  });
 
-      it('log calls are not traced', () => verifyInstanaLoggingIsNotTraced({ expectedAgentLogCount: 1 }));
+  describe('Ensure that Instana logs are not being traced as log spans', function () {
+    beforeEach(async () => {
+      await agentControls.clearReceivedTraceData();
+      await agentControls.clearAgentLogs();
     });
 
-    describe('with a custom pino logger', () => {
-      appControls.registerTestHooks({
-        instanaLoggingMode: 'receives-pino-logger'
+    describe('with the default logger', function () {
+      let appControls;
+
+      beforeEach(async () => {
+        appControls = new AppControls({
+          instanaLoggingMode: 'uses-default-logger'
+        });
+
+        await appControls.start();
+      });
+
+      afterEach(() => {
+        return appControls.stop();
       });
 
-      it('log calls are not traced', () => verifyInstanaLoggingIsNotTraced());
+      it('log calls are not traced', () => {
+        verifyInstanaLoggingIsNotTraced(appControls, { expectedAgentLogCount: 1 });
```

Comment:
The function is being called, but the test case completes before the result is actually returned, which causes a false positive (the test appears to pass even though it hasn’t verified the outcome).

Please ensure the function’s result is returned so that Mocha properly waits for the asynchronous operation to complete. For example:

```suggestion
        return verifyInstanaLoggingIsNotTraced(appControls, { expectedAgentLogCount: 1 });
```

It’s also possible that other test cases have a similar issue, so please review them as well.

Reply:
👍